src/pages/workspace/withPolicyConnections.tsx

```diff
@@ -0,0 +1,69 @@
+import React, {useEffect} from 'react';
+import type {ComponentType} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import FullPageOfflineBlockingView from '@components/BlockingViews/FullPageOfflineBlockingView';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import useNetwork from '@hooks/useNetwork';
+import {openPolicyAccountingPage} from '@libs/actions/PolicyConnections';
+import ONYXKEYS from '@src/ONYXKEYS';
+import withPolicy from './withPolicy';
+import type {WithPolicyProps} from './withPolicy';
+
+type WithPolicyConnectionsProps = WithPolicyProps;
+
+/**
+ * Higher-order component that fetches the connections data and populates
+ * the corresponding field of the policy object if the field is empty. It then passes the policy object
+ * to the wrapped component.
+ *
+ * Use this HOC when you need the policy object with its connections field populated.
+ *
+ * Only the active policy gets the complete policy data upon app start that includes the connections data.
+ * For other policies, the connections data needs to be fetched when it's needed.
+ */
+function withPolicyConnections(WrappedComponent: ComponentType<WithPolicyConnectionsProps>) {
+    function WithPolicyConnections({policy, policyDraft, route}: WithPolicyConnectionsProps) {
+        const {isOffline} = useNetwork();
+        const [hasConnectionsDataBeenFetched, {status}] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY_HAS_CONNECTIONS_DATA_BEEN_FETCHED}${policy?.id ?? ''}` as const, {
```

Comment:
```suggestion
        const [hasConnectionsDataBeenFetched, {status}] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY_HAS_CONNECTIONS_DATA_BEEN_FETCHED}${policy?.id ?? '0'}` as const, {
```
I think you want this, otherwise, you could be getting the entire collection.

Reply:
fixed: https://github.com/Expensify/App/pull/39132/commits/4804fd19f34f2f9c3a89c82b4d6e296d4a1b05ac